README.md

```diff
@@ -21,19 +26,86 @@ of useful vector and large number operations:
   For example vector SIMD implementation for ASCII `__isalpha`, etc.
   Another example full `__int128` implementations of Count Leading Zeros,
   Population Count, and Multiply.
+- provide optimized run-time libraries for quadword integer multiply
+  and multi-quadword integer multiply and add.
 
 ## Build
 
-It is likely sufficient to perform:
+PVECLIB how supports CPU tuned run-time libraries, both static archives
+and dynamic (IFUNC selected) shared objects. This complicates the build
+process as it now has to build; the same source code, multiple times,
+with different compile targets (-mcpu=). Another complication comes
+from compiling for Big Endian systems where the compiler default target
+may not include the vector facilities (VMX and VSX).
+
+## Configure and option flags
+
+The project can use configure test to define options like AM_CPPFLAGS
+and AM_CFLAGS but the user command line options (CPPFLAGS and CFLAGS)
+are always applied last and take precedent.
+See: Automake "Flag Variables Ordering" 
+https://www.gnu.org/software/automake/manual/html_node/Flag-Variables-Ordering.html
+
+So a configure flag like CFLAGS='-O3 -mcpu=power7' would be OK for
+functional verification tests of the POWER7 specific implementations
+of PVECLIB operations. But this would interfere with building the POWER8
+and POWER9 specific compiles for the production version of libpvec.so.
+So builds for production level PVECLIB should never specify -mcpu= in
+CFLAGS.
+
+On the other hand if the user does no specify any CFLAGS, autoconf will
+fill in a default value of '-O2 -g'. This is bad! PVECLIB needs the
+global common subexpression, loop, and vector cost model optimizations
+enable by -O3. Also -g will generate huge debug tables for the vector
+int512 run-time and slow down the build. If you need to profile or
+debug with basic back-trace information, use -g1.
+
+So unless you are involved in the functional testing of new PVECLIB
+operations, the safe options are:
+
+CFLAGS='-m64 -g1 -O3'
+
+The PVECLIB Makefile.am files include special macros for CPU specific
+run-time compiles. These macros exclude the user CFLAGS from those
+compile commands.
+
+On the other hand, if the compiler default target does not support
+PowerISA vector facilities and an appropriate -mcpu= option is not
+supplied, the compile will fail. So the PVECLIB configure.ac includes a
+number of configure tests that detect this and provide appropriate
+compile targets.
+
+The current PVECLIB implementation assumes the target supports both VMX
+(Altivec) and VSX facilities. So the minimum targets are set internally
+(PVECLIB_DEFAULT_CFLAG) to -mcpu=power7 for BE and -mcpu=power8 for LE.
```

Comment:
'-mcpu=power7' and '-mcpu=power8' in single quotes.

Reply:
OK